src/helpers/audio/worker.ts

```diff
@@ -26,6 +26,7 @@ let spectrogram: SpectrogramGenerator;
 let options: SpectrogramOptions;
 
 let state: WorkerState;
+let hasBufferedImage = false;
```

Comment:
i'm in favour of removing this - also doc missing on declaration

Reply:
partly i don't like more global state - especially with the message based flow that happens with this code

Also it seems more elegant to me to do a callback (https://github.com/ecoacoustics/web-components/pull/565#discussion_r2464418625). But that's subjective.